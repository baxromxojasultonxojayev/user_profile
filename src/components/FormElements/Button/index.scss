.button-wrapper {
  .edit {
    padding: 8px 24px;
    background-color: #2489ff;
    color: white;
    border: none;
    cursor: pointer;
    border-radius: 5px;
    &:hover {
      color: red;
      transition: 1s;
    }
    &:active {
      color: blue;
    }
  }
  .cancel {
    padding: 8px 24px;
    background-color: red;
    color: white;
    border: none;
    cursor: pointer;
    border-radius: 5px;
    &:hover {
      color: blue;
      transition: 1s;
    }
    &:active {
      color: blue;
    }
  }
  .save {
    padding: 8px 24px;
    background-color: #2489ff;
    color: white;
    border: none;
    cursor: pointer;
    border-radius: 5px;
    &:hover {
      color: blue;
      transition: 1s;
    }
    &:active {
      color: blue;
    }
  }
}
