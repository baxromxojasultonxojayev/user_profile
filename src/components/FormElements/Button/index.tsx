import React, { FC } from "react";
import "./index.scss";

const Button: FC<Props> = ({ mode, type, onClick }) => {
  return (
    <div className="button-wrapper">
      <button
        onClick={onClick}
        type={type ? type : "text"}
        className={
          mode === "edit" ? "edit" : mode === "cancel" ? "cancel" : "save"
        }
      >
        {mode === "edit"
          ? "O'zgartirish"
          : mode === "cancel"
          ? "Bekor qilish"
          : "Saqlash"}
      </button>
    </div>
  );
};

type Props = {
  mode: string;
  type?: any;
  onClick?: any;
};

export default Button;
