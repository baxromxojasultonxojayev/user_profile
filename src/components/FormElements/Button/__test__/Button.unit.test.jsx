import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"; // Add this import
import Button from "..";

describe("Button Component", () => {
  it("matches snapshot without props", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it("clicking save button", () => {
    const { container, rerender } = render(<Button mode="edit" />);

    rerender(<Button mode="save" />);
    expect(container).toMatchSnapshot();
    expect(
      screen.getByRole("button", { name: /Saqlash/i })
    ).toBeInTheDocument();
  });

  it("clicking cancel button", () => {
    const { container, rerender } = render(<Button mode="cancel" />);

    rerender(<Button mode="cancel" />);
    expect(container).toMatchSnapshot();

    expect(
      screen.getByRole("button", { name: /Bekor qilish/i })
    ).toBeInTheDocument();
  });

  it("clicking edit button", () => {
    const { container, rerender } = render(<Button mode="save" />);

    rerender(<Button mode="edit" />);
    expect(container).toMatchSnapshot();

    expect(
      screen.getByRole("button", { name: /O'zgartirish/i })
    ).toBeInTheDocument();
  });
});
