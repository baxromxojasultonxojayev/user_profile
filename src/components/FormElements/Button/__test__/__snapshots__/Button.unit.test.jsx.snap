// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component clicking cancel button 1`] = `
<div>
  <div
    class="button-wrapper"
  >
    <button
      class="cancel"
      type="text"
    >
      Bekor qilish
    </button>
  </div>
</div>
`;

exports[`Button Component clicking edit button 1`] = `
<div>
  <div
    class="button-wrapper"
  >
    <button
      class="edit"
      type="text"
    >
      O'zgartirish
    </button>
  </div>
</div>
`;

exports[`Button Component clicking save button 1`] = `
<div>
  <div
    class="button-wrapper"
  >
    <button
      class="save"
      type="text"
    >
      Saqlash
    </button>
  </div>
</div>
`;

exports[`Button Component matches snapshot without props 1`] = `
<div>
  <div
    class="button-wrapper"
  >
    <button
      class="save"
      type="text"
    >
      Saqlash
    </button>
  </div>
</div>
`;
